SourceCode/Scene/CRTMesh.h

```diff
@@ -9,6 +11,7 @@ class CRTMesh
 public:
 	CRTMesh(const std::vector<CRTVector>& vertices, const std::vector<int>& triangleVertIndices);
 
-	std::vector<CRTTriangle> getTriangles() const;
+	std::tuple<bool, CRTVector, CRTTriangle> intersectsRay(const CRTRay& ray) const;
```

Comment:
You could use an Intersection data structure for the result of the intersection.